storage/innobase/lock/lock0lock.cc

```diff
@@ -1354,6 +1466,8 @@ lock_rec_create_low(
 	ut_ad(!(type_mode & LOCK_TABLE));
 	ut_ad(trx->state != TRX_STATE_NOT_STARTED);
 	ut_ad(!trx->is_autocommit_non_locking());
+	ut_ad(!c_lock_info.insert_after || !(type_mode & LOCK_WAIT));
+	ut_ad(!c_lock_info.bypassed || c_lock_info.insert_after);
```

Comment:
These could be written as
```c++
ut_ad(c_lock_info.insert_after ? !(type_mode & LOCK_WAIT) : c_lock_info.bypassed);
```
Maybe this could be a member function?
```c++
struct conflicting_lock_info {
// ...
  bool some_predicate_name(unsigned type_mode) const noexcept
  { return insert_after ? (type_mode & LOCK_WAIT) : bypassed; }
};
// ...
  ut_ad(c_lock_info.some_predicate_name(type_mode));
```

Reply:
The clause is used only once for debug build, and `conflicting_lock_info` contains only two members with public access for release build. I don't see any sense in member function for it.